feat: 그룹 상세 화면 지도 탭 구현

HJunng opened the pull request:
## 📝 변경 사항
그룹 상세 화면의 지도 탭을 구현하고, 네이버 지도 SDK를 연동하여 장소 위치를 마커로 표시하는 기능을 추가했습니다.

## 🎯 작업 내용
- 네이버 지도 SDK 연동
  - Naver Map SDK, Compose, Location 라이브러리 추가
  - local.properties에서 API 키 주입 설정
  - 위치 권한 추가
- 지도 탭 구현
  - 장소 위치를 지도에 마커로 표시
  - 커스텀 이미지 마커 컴포넌트 (`ImageMarker`) 구현
  - Coil을 이용한 마커 이미지 미리 로드 및 Bitmap 변환
  - 모든 마커가 보이도록 카메라 자동 조정
- 장소 이미지 BottomSheet 구현
  - 마커 클릭 시 장소 이름과 이미지 목록을 보여주는 BottomSheet 표시
  - Grid 레이아웃으로 이미지 목록 표시
- 에러 처리
  - 지도 로딩 실패 시 에러 화면 표시
  - `ErrorFullScreen` 컴포저블 구현 (재시도 기능 포함)
- UI/UX 개선
  - `HorizontalPager`를 조건부 렌더링으로 변경하여 탭 전환 최적화
  - 다크 모드용 색상 추가 (`DarkOffWhite`, `DarkOnOffWhite`)
- 데이터 모델 구조화
  - `Place` 모델 추가 (장소 정보: 이름, 위경도, 이미지)
  - `ImageItem` 모델 위치 변경 (groupdetail/model → model)
  - 더미 데이터에 장소 데이터 추가

## 🔗 관련 이슈
Closes #19

## 📸 스크린샷 (선택사항)
<img width="345" height="759" alt="image" src="https://github.com/user-attachments/assets/b6b30d69-374a-4f44-8cfc-ada316a141e3" />

<img width="348" height="764" alt="image" src="https://github.com/user-attachments/assets/efe38bab-d2db-46bd-867e-db55119ac802" />



## 💬 리뷰어에게
- 네이버 지도 API 키가 필요합니다. `local.properties`에 `NAVER_MAP_NCP_KEY_ID` 설정해주세욤.
- 마커 이미지는 Coil로 미리 로드하여 Bitmap으로 변환하는 방식을 사용했습니다.
- 지도 로딩이 2초 이상 걸릴 경우 에러로 간주하고 에러 화면을 표시합니다.
- 마커 클릭 시 해당 장소의 이미지를 BottomSheet로 볼 수 있습니다.
- 700줄 죄송함다 끊을 수 없었어요

HJunng committed:
feat: 지도에 표시될 이미지 마커 컴포저블 구현

HJunng committed:
feat: Place 데이터모델 추가 및 PlaceImageItem 확장

- `PlaceImageItem`을 공용 모델인 `ImageItem`으로 이름을 변경하고 `presentation/model` 패키지로 이동
- 장소 정보를 나타내는 `Place` 데이터 클래스를 `presentation/model` 패키지에 추가

HJunng committed:
feat: 네이버 지도 SDK 연동

- 네이버 지도 SDK, Compose, Location 관련 의존성 추가
- 네이버 지도 Maven 저장소를 `settings.gradle.kts`에 추가
- 위치 정보 접근 권한을 `AndroidManifest.xml`에 추가
- `local.properties`에서 네이버 지도 클라이언트 ID를 주입하도록 설정

HJunng committed:
fix: MemoripStaggeredGrid 코드 정리

- `PlaceImageItem`을 범용적인 `ImageItem`으로 변경

HJunng committed:
fix: ImageMarker가 Bitmap을 받도록 수정

- `ImageMarker` 컴포저블이 이미지 URL 대신 Bitmap을 직접 받도록 변경
- 이에 따라 `AsyncImage` 대신 `Image`를 사용하도록 수정

HJunng committed:
feat: 지도 탭에 장소 마커 표시 기능 구현

- Naver Map SDK를 사용하여 지도 표시
- 장소 목록을 기반으로 지도에 마커를 표시
- 마커를 클릭하면 해당 장소의 이미지들을 보여주는 BottomSheet를 노출
- 모든 마커가 보이도록 카메라 위치를 자동으로 조정하는 로직 추가

HJunng committed:
feat: 지도 탭에 장소 마커 표시 기능 구현

- Naver Map SDK를 사용하여 지도 표시
- 장소 목록을 기반으로 지도에 마커를 표시
- 마커를 클릭하면 해당 장소의 이미지들을 보여주는 BottomSheet를 노출
- 모든 마커가 보이도록 카메라 위치를 자동으로 조정하는 로직 추가

HJunng committed:
feat: 장소들을 가져오는 사용될 더미 데이터 추가

- `DummyData` 객체를 추가하여 앱의 여러 화면에서 사용할 임시 데이터를 제공
- `Place`와 `ImageItem` 모델을 기반으로 더미 데이터를 생성
- 이미지 URL은 `picsum.photos`를 활용하여 생성

HJunng committed:
feat: 이미지 마커 테두리 색상 변경

HJunng committed:
feat: 에러 화면 컴포저블 구현

HJunng committed:
feat: 지도 로딩 실패 시 에러 화면 표시

- `NaverMap`의 로딩 상태를 추적하여 2초 내에 로딩되지 않으면 에러로 간주
- 에러 발생 시, `ErrorFullScreen` 컴포저블을 사용해 재시도 버튼과 함께 에러 메시지를 표시
- 지도 로딩 에러에 대한 문자열 리소스 추가

HJunng committed:
feat: GalleryTab이 Place 리스트를 받도록 수정

HJunng committed:
feat: 지도 탭에 장소 마커 표시

- `HorizontalPager`를 제거하고, `selectedTabIndex` 상태를 이용한 조건부 렌더링으로 탭 전환을 구현
- `MapTab`에 장소와 미리 로드한 마커 이미지(`markerImages`)를 전달하여 지도에 마커를 표시하도록 수정
- 마커에 사용할 이미지를 Coil을 이용해 미리 로드하고 `Bitmap`으로 변환하는 로직을 추가

HJunng committed:
feat: 색상 추가

HJunng committed:
feat: 장소 이미지 BottomSheet 구현

- BottomSheet의 콘텐츠로 장소 이름과 이미지 개수를 표시하는 헤더와 이미지 목록을 보여주는 그리드 구현
- 더미데이터 추가

HJunng committed:
Merge remote-tracking branch 'origin/dev' into feature/19-group-detail-map-tab

# Conflicts:
#	client/gradle/libs.versions.toml
#	client/presentation/build.gradle.kts
#	client/presentation/src/main/java/com/andone/memorip/presentation/component/MemoripStaggeredGrid.kt
#	client/presentation/src/main/java/com/andone/memorip/presentation/groupdetail/GroupDetailScreen.kt
#	client/presentation/src/main/java/com/andone/memorip/presentation/groupdetail/component/GalleryTab.kt
#	client/presentation/src/main/java/com/andone/memorip/presentation/util/DummyData.kt

HJunng committed:
feat: PlaceImagesBottomSheet 이미지 코너 Radius 수정 및 더미 데이터 제거

- 불필요해진 더미 데이터 `dummyImages` 삭제

GwonDongHyeon21 commented on client/presentation/src/main/java/com/andone/memorip/presentation/component/ErrorFullScreen.kt:
```diff
@@ -0,0 +1,71 @@
+        Spacer(modifier = Modifier.height(MemoripSpace.SpaceLarge))
+
+        Button(onClick = onRetry) {
+            Text(retryButtonText)
```
파라미터명 추가해주세요!
style도 지정해주세요!

GwonDongHyeon21 commented on client/presentation/src/main/java/com/andone/memorip/presentation/component/MemoripStaggeredGrid.kt:
```diff
@@ -84,11 +84,31 @@ private fun StaggeredImageItem(
+            width = fixedWidth,
+            height = randomHeight
+        )
+    }
```
DummyData 파일로 옮겨두어도 좋을 것 같습니다!

GwonDongHyeon21 commented on client/presentation/src/main/java/com/andone/memorip/presentation/component/ErrorFullScreen.kt:
```diff
@@ -0,0 +1,71 @@
+    MemoripTheme {
+        ErrorFullScreen(
+            onRetry = {}
+        )
```
한줄로 작성해 주세요!

GwonDongHyeon21 commented on client/presentation/src/main/java/com/andone/memorip/presentation/groupdetail/GroupDetailScreen.kt:
```diff
@@ -86,40 +117,37 @@ fun GroupDetailScreenContent(
-                    )
-                    1 -> MapTab(modifier = Modifier.fillMaxSize())
-                }
+            // 선택된 탭에 따라 조건부 렌더링
```
주석은 제거해도 될 것 같습니다!

GwonDongHyeon21 commented on client/presentation/src/main/java/com/andone/memorip/presentation/groupdetail/component/PlaceImagesBottomSheet.kt:
```diff
@@ -0,0 +1,156 @@
+import com.andone.memorip.presentation.util.DummyData
+
+private object PlaceImagesBottomSheetDimens {
+    const val ImageAspectRatio = 1f
```
constants 관련 object로 나누는게 어떤가요?

GwonDongHyeon21 commented on client/presentation/src/main/java/com/andone/memorip/presentation/groupdetail/component/MapTab.kt:
```diff
@@ -1,20 +1,181 @@
+
+@OptIn(ExperimentalNaverMapApi::class)
+@Composable
+private fun PlaceMarkers(
```
별도의 component로 분리하는 거 어떤가요?

GwonDongHyeon21 reviewed (commented):
고생하셨습니다. 지도 만들어두신거 참고해서 구현해보겠습니다!

GwonDongHyeon21 commented on client/presentation/src/main/java/com/andone/memorip/presentation/groupdetail/component/MapTab.kt:
```diff
@@ -1,20 +1,181 @@
+                }
+            ) {
+                ImageMarker(imageBitmap = bitmap)
+            }
```
PlaceMarker라는 이름이 있어서 ImageMarker보고 Marker 종류가 두개인가? 순간 생각이 들었습니다. 이름 수정하시는 거 어떠신가요?

HJunng replied:
ImageMarker를 공용으로 재사용할 수 있다고 생각해서 공용 컴포넌트로 분리했고, 
PlaceMarkers를 PlacesSection으로 할까 하다가 이미지 마커를 놓는 용도인게 명확하게 보이는게 좋을 것 같아서PlaceImageMarkers로 Place에 대한 ImageMarker들을 찍는 컴포넌트로 해서 이름 변경해두었습니다!

HJunng committed:
style: ErrorFullScreen 컴포저블의 버튼 스타일 수정

HJunng committed:
style: ErrorFullScreen 컴포저블의 버튼 스타일 수정

HJunng committed:
chore: 더미 이미지 데이터 분리

- 더미 이미지 생성 함수 `generateRandomImageUrls`를 `DummyData` 객체로 이동

HJunng committed:
style: 그룹 상세 화면 및 장소 이미지 BottomSheet 코드 정리

- `GroupDetailScreen.kt`에서 불필요한 주석 제거
- `PlaceImagesBottomSheet.kt`의 상수를 별도 객체로 분리

HJunng committed:
feat: ImageMarker 공용 컴포넌트로 분리

HJunng committed:
chore: `PlaceImageMarkers` 컴포넌트 분리

- `MapTab` 내부에 있던 장소 마커 표시 로직을 `PlaceImageMarkers` 로 분리

hoyadong1 reviewed (approved):
고생하셨슴다. 다크모드 대응 해서 색상 추가 하신거 좋네요 저도 대응 안된 부분 추가해야겠네요
